File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        rechable = 0
        for i,num in enumerate(nums):
            if i>rechable:return False
            if rechable > len(nums)-1:return True
            rechable = max(rechable, i+num)
        return True

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
       //keep track of how many steps you can take forward
       //if current step is greater than, your reachable , give false
       //if current reachable is greater than total length , return true
       //update reachable 
        int reachable = 0;
       for(int i=0;i<nums.length;i++){

        if(i > reachable) return false;

        if(nums.length-1 <= reachable ) return true;

        reachable = Math.max(reachable, i+nums[i]);

       }
       return false;
    }

}